mintpy/objects/cluster.py

```diff
@@ -328,9 +329,10 @@ def collect_result(self, futures, results, box, submission_time):
                     results[i][y0:y1, x0:x1] = sub_result
 
                 else:
-                    msg = "worker result has unexpected dimension: {}".format(num_dim)
-                    msg += '\nit should be either 2 or 3 or 4!'
-                    raise Exception(msg)
+                    if i != 1:
+                        msg = "worker result has unexpected dimension: {}".format(num_dim)
+                        msg += '\nit should be either 2 or 3 or 4!'
+                        raise Exception(msg)
```

Comment:
I don't get the rationale behind this change. Could you explain or better, comment in the code?

Reply:
@mirzaees 
I'm also trying to understand this part. Do you mean that when i==1, that is when the sub_result represents covariance matrix?
Is that because the ```func``` that we pass in, has always a form of :

X, covX, ... = func(some inputs)

?

Reply:
@hfattahi and @yunjunz 
You are right, the logic is not always valid
It is correct for ifgram inversion but not other places, dem_erro.py is also using dask with different outputs. I will modify this in a way that 'ifgram_inversion_patch' function does not return cov matrix if calc_cov is False

Reply:
I adjusted the code to check None in the output so that the DaskCluster object could support None also.